islands/form/tag-input.tsx

```diff
@@ -64,21 +63,38 @@ export default function TagInput<T extends HasLabel>(
   const handleOptionClick = (option: T) => {
     handleSelect(option);
     setInputValue("");
+    setIsFocused(false);
+  };
+
+  const handleInputFocus = () => {
+    setIsFocused(true);
+  };
+
+  const handleInputBlur = () => {
+    !shouldShowOptions && setIsFocused(false);
```

Comment:
I think that this should not depend on `shouldShowOptions`, since `isFocused` is an independent state

Reply:
Nice catch, I will fix that in my next PR, I just wanted to clarify that if we only remove `!shouldShowOptions`, the selection won't take effect any more since the options div will disappear before getting the info from it. 
Though, I just noticed that since the hiding logic is triggered by the mouseLeave event now, we can remove the onBlur logic.